imperial_coldfront_plugin/gpfs_client.py

```diff
@@ -255,6 +255,35 @@ def create_fileset_directory(
                 return task_data
             raise
 
+    @check_job_status
+    @json
+    @get("filesystems/{filesystemName}/filesets/{filesetName}/quotas")
+    def _retrieve_quota_usage(
+        self,
+        filesystemName: str,
+        filesetName: str,
+    ) -> requests.Response:
+        """Method (private) to retrieve the quota usage of a fileset."""
+        pass
+
+    def retrieve_quota_usage(
+        self,
+        filesystem_name: str,
+        fileset_name: str,
+    ) -> requests.Response:
+        """Method (public) to retrieve the quota usage of a fileset.
+
+        Args:
+            filesystem_name: Name of the filesystem where the quota will be set.
+            fileset_name: Name of the fileset to set the quota.
+
+        Returns:
+            The response after successfully retrieving the quota usage.
+        """
+        return self._retrieve_quota_usage(
+            filesystemName=filesystem_name, filesetName=fileset_name
+        )
```

Comment:
It would be nice to do some processing of the return data so that we only get the block and files quota values.